src/gen/conditions.rs

```diff
@@ -1195,7 +1216,7 @@ fn test_coin_id(parent_id: &[u8; 32], puzzle_hash: &[u8; 32], amount: u64) -> By
 fn parse_list_impl(
     a: &mut Allocator,
     input: &str,
-    callback: &Option<fn(&mut Allocator) -> NodePtr>,
+    callback: &Option<Box<dyn Fn(&mut Allocator) -> NodePtr>>,
```

Comment:
this changes a plain function pointer to type-erased function object. This allows passing in closures as well as plain functions. The new test requires capturing `count` and `cond`